pyuvdata/tests/test_uvflag.py

```diff
@@ -18,21 +19,81 @@
 from pyuvdata import version as uvversion
 import shutil
 import copy
+import warnings
+
+pytest_cases = pytest.importorskip('pytest_cases')
```

Comment:
This was my first work-around to skipping if a user doesn't have the plugin. The plugin is used to make decorators for tests in order to parameterize. At first glance I wasn't sure how to set this up to only skip if this plugin cannot be imported, but still have the decorators if it is installed. Maybe it would require some meta-decorator? or I am thinking too hard about it.

Reply:
Maybe we can get @plaplant to weigh in, he might know. A really simple option would be to split the tests into 2 files, but that might cause other problems if there are functions you use in both kinds of tests.

Reply:
@mkolopanis can you define a decorator like `skipIf_no_h5py` and apply that before the decorators for parameterizing? Not sure if that would work, but might be worth a try.

Reply:
Yeah this definitely looks possible. Looks like it would require a little funny looking code though where not `pytest-cases` fixtures are defined directly after the bulk of the imports, then a try/catch block to see if pytest_cases works 
```
@pytest.fixture(scope='function')
def uvf_from_miriad():
    uv = UVData()
    uv.read_miriad(test_d_file)
    uvf = UVFlag(uv)

    # yeild the object for the test
    yield uvf

    # do some cleanup
    del(uvf, uv)


@pytest.fixture(scope='function')
def uvf_from_uvcal():
    uvc = UVCal()
    uvc.read_calfits(test_c_file)
    uvf = UVFlag(uvc)

    # yeild the object for the test
    yield uvf

    # do some cleanup
    del(uvf, uvc)


@pytest.fixture(scope='function')
def uvf_from_waterfall():
    uv = UVData()
    uv.read_miriad(test_d_file)
    uvf = UVFlag(uv, waterfall=True)

    # yeild the object for the test
    yield uvf

    # do some cleanup
    del(uvf, uv)


try:
    import pytest_cases

    cases_decorator = pytest_cases.pytest_parametrize_plus(
        "input_uvf", [pytest_cases.fixture_ref(uvf_from_miriad),
                      pytest_cases.fixture_ref(uvf_from_uvcal),
                      pytest_cases.fixture_ref(uvf_from_waterfall)])

    cases_decorator_no_waterfall = pytest_cases.pytest_parametrize_plus(
        "input_uvf", [pytest_cases.fixture_ref(uvf_from_miriad),
                      pytest_cases.fixture_ref(uvf_from_uvcal)])
except ImportError:
    cases_decorator = uvtest.skipIf_no_pytest_cases
    cases_decorator_no_waterfall = uvtest.skipIf_no_pytest_cases
```

Reply:
I think that's reasonable, maybe just add a comment to explain what's going on if it doesn't seem self explanatory.

I'll note that in the code you pasted, `yield` is spelled wrong in several comment lines.

Reply:
yes, yes it is. I'll comment up around this too. help it make more sense

Reply:
Should I remove `pytest-cases` from the min_deps environment yaml too? Feels like since we have a workaround it's not really appropriate for min_deps anymore.

Reply:
I'm not sure. The min_deps was really to test optional code dependencies, not optional test dependencies, but on the other hand, the primary goal is to make sure the tests run and don't error if you don't have all the dependencies.